### PR TITLE
[4.0] TinyMCE & Codemirror updates with new lineheight for tinymce

### PR DESCRIPTION
**Pull request by brianteeman**
This PR updates Tinymce and codemirror to the latest releases

For codemirrorir just fixes some bugs (https://codemirror.net/doc/releases.html)

TinyMCE has a few new possibilities (https://www.tiny.cloud/docs/release-notes/release-notes55/) and I have implemented the new lineheight button and added it to the advanced preset

To apply the pr you will need to run npm i

In the tinmymce editor plugin go to set 0 and click on use advanced preset
You will now see the new button
image

And / Or drag the new button from the list of available buttons
image

And then test it in the editor itself
image

image

### CODE REVIEW

**Comment by dgrammatiko** [comment]
@brianteeman it's really disturbing that Joomla's tinyMCE integration still allows inline styles. Not that I didn't try to change that: https://github.com/joomla/joomla-cms/pull/14456

**Comment by brianteeman** [comment]
The inline style is from tinymce, not our integration. Unless I did something very wrong.

**Comment by adj9** [comment]
I have tested this item :white_check_mark: successfully on ff4fa779486de6598e527df61a97601699347cbf<br /><br />I tried to install the PR with Joomla beta6. After the installation, I tested the plugin and the use of TinyMCE.

The new bar is 
Schermata 2020-11-01 alle 14 59 33

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31283">issues.joomla.org/tracker/joomla-cms/31283</a>.</sub>

**Comment by zero-24** [comment]
@brianteeman it seens the last commits now includes unintended changes?

**Comment by brianteeman** [comment]
DOH

**Comment by brianteeman** [comment]
@zero-24 should be ok now

**Comment by zero-24** [comment]
Looks good (I'm not sure about the other changes in the package.json are expected but i know to less about that kind of things) The files / commits i saw earlier are now gone. Thanks!

File: plugins/editors/tinymce/tinymce.php [hunk 1]
```diff
@@ -934,6 +934,7 @@ public static function getKnownButtons()
 			'aligncenter'   => array('label' => 'Align center'),
 			'alignright'    => array('label' => 'Align right'),
 			'alignjustify'  => array('label' => 'Justify'),
+			'lineheight'     => array('label' => 'Line height'),
```
Comment (Quy) [inline plugins/editors/tinymce/tinymce.php]:
```suggestion
			'lineheight'    => array('label' => 'Line height'),
```

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on b284789063a5bff12d151d389ef1f4d3ed7e9f80<br /><br />Confirmed new icon is available as well as that it functions as expected.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31283">issues.joomla.org/tracker/joomla-cms/31283</a>.</sub>

**Comment by gostn** [comment]
I have tested this item :white_check_mark: successfully on b284789063a5bff12d151d389ef1f4d3ed7e9f80<br /><br />Test by using https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/31283/downloads/37271/pr_list.xml<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31283">issues.joomla.org/tracker/joomla-cms/31283</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31283">issues.joomla.org/tracker/joomla-cms/31283</a>.</sub>

**Comment by wilsonge** [comment]
@brianteeman need you to resolve the conflicts here with your other PR I merged

**Comment by brianteeman** [comment]
Should be correct now. Knew that was going to happen but no way to avoid it

**Comment by laoneo** [comment]
Thanks!

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**